rfc/20240529-OpenTofu-Specific-Code-Override.md

```diff
@@ -0,0 +1,161 @@
+2. if someone starts a new project (tofu only), do we recommend going with `.tofu` files?
+TODO: List questions that should be discussed and answered during the RFC process.
+
+## Cons
+A major drawback of this solution is that it leads to code duplication for users who need to stay compatible with both OpenTofu and Terraform (such as module maintainers). If These users want to use an OpenTofu-specific feature, they must create and manage two copies of the same file, one with the `.tf` extension and one with the `.tofu` extension. Potentially, in the future, they might have multiple duplicated files in their project. It can be confusing and error-prone, and might cause configuration drifts between the Terraform and OpenTofu configuration over time.
+
+## Future Considerations
+1. Maybe in the far future, when migration from Terraform to OpenTofu might impose a challenge on our users, we can add a command/tool that automatically converts `.tf` files to `.tofu` files in the project. This way, users will have to put a very small effort in migrating to OpenTofu. A good example for a similar tool is the [Java to Kotlin Converter](https://kotlinlang.org/docs/mixing-java-kotlin-intellij.html#converting-an-existing-java-file-to-kotlin-with-j2k).
+2. I don't think we need it now, maybe when we'll have more OpenTofu-specific feature, but we can create Best Practices around how to work with `.tofu` files.
+
+## Potential Alternatives
+1. One proposition was that the `.tofu` file will not override the `.tf` file completely, but will instead override specific blocks. This way, we can avoid code duplication. But this solution is not sufficient for the future, when Terraform might introduce features that OpenTofu cannot parse.
+2. We thought about using a different file extension, like `.otf`, but we decided to use `.tofu` because the `.otf` extension is already used to represent [OpenType](https://en.wikipedia.org/wiki/OpenType) font files.
+
+## General Break Down to Tasks
+In my opinion, this is an opportunity to standardize the way IDEs and 3rd party tools support OpenTofu. So, we should not only pursue technical tasks, but also include peripheral tasks to establish a better experience for our users. Here are the tasks that should be done as part of this RFC:
+1. Approve this RFC by the core-team and the community.
+2. Implement the technical change. Including tests and documentation.
+3. IDEs:
+   * Approach the maintainers of the OpenTofu VSCode Extension and OpenTofu Language Server, notify about this change, and see if we can help.
+   * Decide on a solution for the JetBrains IDEs plugin.
+4. 3rd party tools:
+   * Notify all relevant impacted tools maintainers about this change.
+   * Consider if we want to approach maintainers of tools that don't officially support OpenTofu and see if they are willing to support OpenTofu.
+
+## Issues waiting for this RFC
```

Comment:
I think we should add version compatibility issue here as well: https://github.com/opentofu/opentofu/issues/1708

Reply:
https://github.com/opentofu/opentofu/pull/1699/commits/70554ab32da77a3b2a2098c6070bf9d2f105c9be